### ISSUE TITLE
Migrate multiple/custom runner labels before deprecation

### ISSUE THREAD

**Issue by ZainRizvi** [pytorch/pytorch#125721]
Request from github. These appear to be only on domain libraries: https://github.com/search?type=code&q=-core-ubuntu+user%3Apytorch

#### TL/DR:
Github will be migrating all customers who are currently still using multiple/custom labels on their Larger GitHub Hosted runners to a single label by the end of May 2024.

Swap:
- 8-core to  8-core-ubuntu 
- 16-core to 16-core-ubuntu 

#### Action to take
Delete and re-create runners which have multiple labels, replacing them with runners where they target the runner name by the end of May 2024.

Labels:
- 8-core
- 16-core
where these are 'additional' labels on runners.

So swap:
- 8-core to  8-core-ubuntu 
- 16-core to 16-core-ubuntu 

This changes how the system queues jobs not where those jobs end up

More details: https://github-partners.slack.com/archives/C01KLRNL8NN/p1715015597452249

cc @seemethere @malfet @pytorch/pytorch-dev-infra

**Comment by malfet** [comment]
Done
